packages/gatsby-plugin-page-creator/src/create-client-only-page.ts

```diff
@@ -9,12 +13,14 @@ import { getMatchPath } from "gatsby-core-utils"
 export function createClientOnlyPage(
   filePath: string,
   absolutePath: string,
-  actions: Actions
+  actions: Actions,
+  trailingSlash: TrailingSlash
 ): void {
   const path = createPath(filePath)
+  const modifiedPath = applyTrailingSlashOption(path, trailingSlash)
```

Comment:
Add e2e test for src/pages/client-only/[...splat].js and then see if createPath needs third arg